tests/io.rs

```diff
@@ -43,10 +49,39 @@ fn read_file() {
     }
 }
 
+#[test]
+fn write_and_flush_file() {
+    let _ = ::env_logger::try_init();
+
+    let file =  NamedTempFile::new().unwrap();
+
+    let thread = new_vm();
+    let text = format!(
+        r#"
+        let {{ assert }} = import! std.test
+        let {{ OpenOptions, open_file_with, write_slice_file, flush_file, ? }} = import! std.io
+
+        do file = open_file_with "{path}" [Write]
+        do bytes_written = write_slice_file file [1b, 2b, 3b, 4b] 0 4
+        assert (bytes_written == 4)
+        flush_file file
+        "#,
+        path = file.path().to_str().unwrap()
```

Comment:
I find it a bit cleaner (but slightly more verbose I suppose) to return a function which takes the path as an argument instead of injecting the path via a format string.

```
\path ->
    do file = open_file_with "{path}" [Write]
    ...

// ... 
run_expr::<OwnedFunction<fn (String) -> IO<()>>>(&thread, "<top>", &text)

f.call(file.path().to_str().unwrap()).unwrap();
```

Reply:
You're absolutely right, for some reason I didn't even think of that. Was probably me unconsciously being lazy :p